server/lib/reformat-salesforce-contract.js

```diff
@@ -16,46 +17,39 @@ module.exports = exports = SFContract => {
 		assets: []
 	};
 
-	contract.assets = SFContract.assets.filter(({ assetType }) => assetType !== 'Addendum').map(item => {
-		const asset = formatAsset(item);
+	contract.assets = SFContract.assets
+		.filter(({assetType}) => assetType !== 'Addendum')
+		.map(formatAsset)
+		.map(asset => {
+			asset.download_limit = SFContract[`${asset.content_type}Limit`];
+			return asset;
+		});
 
-		asset.download_limit = SFContract[`${asset.content_type}Limit`];
-
-		return asset;
-	});
-
-	SFContract.assets.filter(({ assetType }) => assetType === 'Addendum').forEach(item => {
-		const addendum = formatAsset(item);
-
-		let asset = contract.assets.find(asset =>
-			asset.content_type === addendum.content_type
-			&& asset.content_set === addendum.content_set);
-
-		if (!asset) {
-			asset = contract.assets.find(asset =>
-				asset.content_type === addendum.content_type);
+	SFContract.assets
+		.filter(({assetType}) => assetType === 'Addendum')
+		.map(formatAsset)
+		.forEach(addendum => {
+			const asset = findMatchingAsset(addendum, contract);
 
 			if (!asset) {
-				throw new ReferenceError(`Asset not found for Addendum: ${JSON.stringify(addendum)}`);
+				logger.error(new ReferenceError(`Asset not found for Addendum: ${JSON.stringify(addendum)}`));
```

Comment:
Looks good. Thought that @leggsimon might have some opinions on, would it be good to set up a splunk alert based on this?

Reply:
Yes it would, or logging to slack or something